cli/src/commands/next.rs

```diff
@@ -56,9 +56,15 @@ pub(crate) struct NextArgs {
     offset: u64,
     /// Instead of creating a new working-copy commit on top of the target
     /// commit (like `jj new`), edit the target commit directly (like `jj
-    /// edit`).
-    #[arg(long, short)]
+    /// edit`). Takes precedence over config in `ui.movement.edit`; i.e.
+    /// will negate `ui.movement.edit = "never"`
+    #[arg(long, short, conflicts_with = "no_edit")]
     edit: bool,
+    /// The inverse of `--edit`.
+    /// Takes precedence over config in `ui.movement.edit`; i.e.
+    /// will negate `ui.movement.edit = "always"`
+    #[arg(long, short, conflicts_with = "edit")]
+    no_edit: bool,
```

Comment:
Another option is to allow `--edit` and `--edit=true|false`, but I don't know if we like it.
https://docs.rs/clap/latest/clap/struct.Arg.html#method.default_missing_value

This will probably apply to `git init --no-colocate`.

Reply:
I'm not sure I like it either, but an advantage of that style is that we can presumably extend it later to support e.g. `--edit=auto` (I don't want that particular option; I'm just saying that it's useful in general).

Reply:
So today we have the following states:

```
--edit, --no-edit - illegal
--edit,   --      - edit mode
   -- , --no-edit - new+squash mode
   -- ,   --      - get mode from config
```

if we go with a purely boolean flag, the config will have no effect:

```
--edit=true                 - edit mode
--edit=false                - new+squash mode
--edit (implies true)       - edit mode
unspecified (implies false) - new+squash mode
```

What we'll actually want is `--edit=true|false|config` with `default_value="config", default_missing_value="true"`
```

--edit=true                 - edit mode
--edit=false                - new+squash mode
--edit (implies true)       - edit mode
unspecified (implies config)- get mode from config
```

I personally find this tri-state, but mostly boolean flag to be weird and surprising behaviour. `--edit |--no-edit` is more expected behaviour.

That said, if the consensus is `--edit=true|false|config`, I'll go ahead to implement that.

Reply:
I still think you are describing what 1509 says to avoid -- commands behaving differently for different people because the config changes the default.

Reply:
> I still think you are describing what 1509 says to avoid -- commands behaving differently for different people because the config changes the default.

I do think there's an important difference. #1509 says not to allow making arbitrary cli flags into defaults as that leads differing behaviours for different users. The reason for that is because anyone can make a flag that should be explicitly enabled into an implicit default.

This PR doesn't do that. It adds a very narrow and specific default that is available for everyone, and importantly, provides the same behaviour, if the same config value is set by different users.

Reply:
> --edit=true                 - edit mode
> --edit=false                - new+squash mode
> --edit (implies true)       - edit mode
> unspecified (implies config)- get mode from config

I was hoping that `--edit` (without a value) would be disallowed. `--edit=` with an empty string, or anything that's not "true", "false", or "config", would be an error. So I think that means a string-valued flag with a default of "config". Is that possible? Would you find that behavior clearer?

Reply:
Unless we need an explicit `--edit=<USE_CONFIG_VALUE>`, it can be implemented as `Option<bool>` argument.

FWIW, I found `jj new` has `--no-edit`, so `--edit`/`--no-edit` might make sense.

Reply:
True, we already have some `--no-*` flags, so `--no-edit` seems least surprising then.

Reply:
Nice. So I'll leave it with `--edit|--no-edit` then. Resolving this. Do re-open if I've missed something. Thanks.